clients/include/testing_tbmv_batched.hpp

```diff
@@ -0,0 +1,236 @@
+/* ************************************************************************
+ * Copyright 2018-2019 Advanced Micro Devices, Inc.
+ *
+ * ************************************************************************ */
+
+#include "cblas_interface.hpp"
+#include "flops.hpp"
+#include "near.hpp"
+#include "norm.hpp"
+#include "rocblas.hpp"
+#include "rocblas_datatype2string.hpp"
+#include "rocblas_init.hpp"
+#include "rocblas_math.hpp"
+#include "rocblas_random.hpp"
+#include "rocblas_test.hpp"
+#include "rocblas_vector.hpp"
+#include "unit.hpp"
+#include "utility.hpp"
+
+template <typename T>
+void testing_tbmv_batched_bad_arg(const Arguments& arg)
+{
+    const rocblas_int M           = 100;
+    const rocblas_int K           = 5;
+    const rocblas_int lda         = 100;
+    const rocblas_int incx        = 1;
+    const rocblas_int batch_count = 5;
+
+    const rocblas_fill      uplo   = rocblas_fill_upper;
+    const rocblas_operation transA = rocblas_operation_none;
+    const rocblas_diagonal  diag   = rocblas_diagonal_non_unit;
+
+    rocblas_local_handle handle;
+
+    size_t size_A = lda * size_t(M);
+    size_t size_x = M * size_t(incx);
+
+    // allocate memory on device
+    device_vector<T*, 0, T> dA(batch_count);
+    device_vector<T*, 0, T> dx(batch_count);
+    if(!dA || !dx)
+    {
+        CHECK_HIP_ERROR(hipErrorOutOfMemory);
+        return;
+    }
+
+    EXPECT_ROCBLAS_STATUS(
+        rocblas_tbmv_batched<T>(
+            handle, uplo, transA, diag, M, K, nullptr, lda, dx, incx, batch_count),
+        rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        rocblas_tbmv_batched<T>(
+            handle, uplo, transA, diag, M, K, dA, lda, nullptr, incx, batch_count),
+        rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(
+        rocblas_tbmv_batched<T>(nullptr, uplo, transA, diag, M, K, dA, lda, dx, incx, batch_count),
+        rocblas_status_invalid_handle);
+}
+
+template <typename T>
+void testing_tbmv_batched(const Arguments& arg)
+{
+    rocblas_int       M           = arg.M;
+    rocblas_int       K           = arg.K;
+    rocblas_int       lda         = arg.lda;
+    rocblas_int       incx        = arg.incx;
+    char              char_uplo   = arg.uplo;
+    char              char_diag   = arg.diag;
+    rocblas_fill      uplo        = char2rocblas_fill(char_uplo);
+    rocblas_operation transA      = char2rocblas_operation(arg.transA);
+    rocblas_diagonal  diag        = char2rocblas_diagonal(char_diag);
+    rocblas_int       batch_count = arg.batch_count;
+
+    rocblas_local_handle handle;
+
+    // argument sanity check before allocating invalid memory
+    if(M < 0 || K < 0 || lda < M || lda < 1 || !incx || K >= lda || batch_count <= 0)
+    {
+        static const size_t     safe_size = 100; // arbitrarily set to 100
+        device_vector<T*, 0, T> dA1(safe_size);
```

Comment:
Consider device_batch_vector ?

Reply:
Ah right, I forgot about the device_vector refactoring. Changed based off of testing_axpy_batched.hpp.